test/tools/router_check/test/config/DirectResponse.golden.proto.json

```diff
@@ -1,14 +1,104 @@
 {
-    "tests":  [
-      {
-        "test_name": "Direct Response",
-        "input": {
-          "authority": "lyft.com",
-          "path": "/ping",
-          "method": "GET"
-        },
-        "validate": {"path_redirect": "http://lyft.com/ping"}
-      }
-    ]
-  }
-  
\ No newline at end of file
+  "tests": [
+    {
+      "test_name": "Direct Response",
+      "input": {
+        "authority": "lyft.com",
+        "path": "/ping",
+        "method": "GET"
+      },
+      "validate": {
+        "path_redirect": "http://lyft.com/ping",
+        "cluster_name": "",
+        "host_rewrite": "",
+        "path_rewrite": "",
+        "virtual_cluster_name": "",
+        "virtual_host_name": "",
+        "response_header_fields": [
+          {
+            "key": "content-type",
+            "value": "text/plain"
+          },
+          {
+            "key": "content-length",
+            "value": "8"
+          }
+        ]
+      }
+    },
+    {
+      "test_name": "Direct Response custom content-type",
+      "input": {
+        "authority": "lyft.com",
+        "path": "/ping-json",
+        "method": "GET"
+      },
+      "validate": {
+        "path_redirect": "http://lyft.com/ping-json",
+        "cluster_name": "",
+        "host_rewrite": "",
+        "path_rewrite": "",
+        "virtual_cluster_name": "",
+        "virtual_host_name": "",
+        "response_header_fields": [
+          {
+            "key": "content-type",
+            "value": "application/json"
+          },
+          {
+            "key": "content-length",
+            "value": "25"
+          }
+        ]
+      }
+    },
+    {
+      "test_name": "Direct Response custom content-type appended",
+      "input": {
+        "authority": "lyft.com",
+        "path": "/ping-json2",
+        "method": "GET"
+      },
+      "validate": {
+        "path_redirect": "http://lyft.com/ping-json2",
+        "cluster_name": "",
+        "host_rewrite": "",
+        "path_rewrite": "",
+        "virtual_cluster_name": "",
+        "virtual_host_name": "",
+        "response_header_fields": [
+          {
+            "key": "content-type",
+            "value": "text/plain,application/json"
+          }
+        ]
+      }
+    },
+    {
+      "test_name": "Direct Response empty body",
+      "input": {
+        "authority": "lyft.com",
+        "path": "/ping-empty",
+        "method": "GET"
+      },
+      "validate": {
+        "path_redirect": "http://lyft.com/ping-empty",
+        "cluster_name": "",
+        "host_rewrite": "",
+        "path_rewrite": "",
+        "virtual_cluster_name": "",
+        "virtual_host_name": "",
+        "response_header_fields": [
+          {
+            "key": "content-type",
+            "value": ""
```

Comment:
Is this goal here to test for absence? It would be a much stronger test if we had an explicit way of testing for this?

Reply:
Yes the goal is to test for absence of the content-type header. 
Looking at https://tools.ietf.org/html/rfc7230#section-3.2.6, it is possible for field values to be implicitly empty strings (whitespace (trimmed) + CRLF), or explicitly double-quoted empty strings.
It would additionally be stronger from a human readable perspective to be able to say, for instance:
```
validate:
  response_header_fields:
    - key: content-type
      absent: true
```

Reply:
I filed https://github.com/envoyproxy/envoy/issues/10229 for future improvement.

Reply:
Let's please just fix this now. Sorry I should have commented on this in @LisaLudique's last PR, but since we just switched the proto to split request/response headers for matching, we should use the `HeaderMatcher` proto that we already have which makes this type of matching easily possible. Thank you!

Reply:
I don't like the scope creep, but ok.
What's the policy on proto changes? Do I need to rename the fields and reserve the old field numbers or am I ok to change it inline because there wasn't a version cut in between?

Reply:
We are being a bit loose on this tool, so I think it's OK to just change it since we just changed it already within the last week. I would talk to @LisaLudique about it.